services/py-api/src/utils.py

```diff
@@ -1,8 +1,10 @@
 from threading import Lock
```

Comment:
Hey, hey @plamenapavlova @IvanObreshkov @Isak-Bego . This class seems to be really coupled with the hackathon business logic. What if you decide to use it for an admin panel where hubbers should use jwt to authenticate and authorize themselves? Will their tokens use the same schema?

```
            if "sub" in decoded_token and "is_admin" in decoded_token and "team_id" in decoded_token:
                return Ok(cast(JwtUserData, decoded_token))

            LOG.warning(
                "Decoded token is missing some or all of the required fields: `sub`, `is_admin`, `team_id`",
                decoded_token=decoded_token,
            )
```

Reply:
The check is here only due to the schema. You are right that we have to design it to be more generic, but we have to come up with a way to return TypedDIcts or better create custom types for the different use cases. And just pass them as Generics or something like that.